plugins/module_utils/_api_data.py

```diff
@@ -13,18 +13,20 @@
 class APIData(object):
     def __init__(self, primary_keys=None,
                  stratify_keys=None,
+                 require_one_of=None,
                  has_identifier=False,
                  single_value=False,
                  unknown_mechanism=False,
                  fully_understood=False,
                  fixed_entries=False,
                  fields=None):
-        if sum([primary_keys is not None, stratify_keys is not None, has_identifier, single_value, unknown_mechanism]) > 1:
-            raise ValueError('primary_keys, stratify_keys, has_identifier, single_value, and unknown_mechanism are mutually exclusive')
+        if sum([primary_keys is not None, stratify_keys is not None, require_one_of is not None, has_identifier, single_value, unknown_mechanism]) > 1:
```

Comment:
I don't think `require_one_of` should be mutually exclusive with the others. Also it should be a list of lists, similar to the Ansible module argument spec validation functionality, not a single list of strings.

Reply:
I agree on the mutual exclusivity, I'll revert that.

As for list of lists, do you mean something where:

```py
require_one_of = [["a1", "b1"], ["a2", "b2"]]
```

would require you provide ("a1" OR "b1") AND ("a2" OR "b2")?

Apologies, I am not very familiar with Ansible's internals just yet.

Reply:
Yes, that's what I mean. (The lists could also have more than two elements though.)

Reply:
I've changed the type to a list of lists and added some tests.